### PR TITLE
Add CSRF protection to "/logout"

### LINKED ISSUE #33030: Add CSRF protection to "/logout"
### Body

The `/logout` endpoint has no CSRF protection. This is not a security issue, because the user is logged out when the CSRF protection is in-place, but it is a potential issue as the user might get logged out by a 3rd-party tab opened (potentially), 

Would be great to take a look and fix it.

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by shahar1**
closes: #33030

cc: @potiuk, @eladkal

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by shahar1** [comment]
I'd be happy for some minor UI help with alignment and color (my CSS abilities are good as my french 😅):

image

File: airflow/www/templates/appbuilder/navbar_right.html [hunk 1]
```diff
@@ -83,7 +83,12 @@
         <li><a href="{{user_profile_url}}"><span class="material-icons">account_circle</span>{{_("Your Profile")}}</a></li>
         <li role="separator" class="divider"></li>
       {% endif %}
-      <li><a href="{{auth_manager.get_url_logout()}}"><span class="material-icons">exit_to_app</span>{{_("Log Out")}}</a></li>
+      <li class="dropdown">
+        <form method="POST" action="{{auth_manager.get_url_logout()}}">
+        <input type="hidden" name="csrf_token" value="{{ csrf_token() }}"/>
+        <button type="submit" class="btn btn-link"><span class="material-icons">exit_to_app</span>{{_("Log Out")}}</button>
```
Comment (shahar1) [inline airflow/www/templates/appbuilder/navbar_right.html:89]: I'd be happy for help with alignment and color (my CSS abilities are good as my french 😅):

image

**Comment by shahar1** [comment]
General question - Should Non-DB tests for FAB provider be skipped? (currently they fail)

**Comment by potiuk** [comment]
> General question - Should Non-DB tests for FAB provider be skipped? (currently they fail)

I think by modifying logout and adding decorator, some of the tests simply started requiring database - have not looked in detail, but that's essentially what is happening here. I guess it can be fixed by changing the test code to not trigger DB operations in those tests (preferrable) or marking them as db_tests.

File: airflow/providers/fab/auth_manager/security_manager/override.py [hunk 2]
```diff
@@ -57,8 +57,10 @@
     AuthOAuthView,
     AuthOIDView,
     AuthRemoteUserView,
+    AuthView,
     RegisterUserModelView,
 )
+from flask_appbuilder.views import expose
```
Comment (potiuk) [inline airflow/providers/fab/auth_manager/security_manager/override.py:63]: I guess this import is causing the DB access.

**Comment by shahar1** [comment]
> > General question - Should Non-DB tests for FAB provider be skipped? (currently they fail)
> 
> I think by modifying logout and adding decorator, some of the tests simply started requiring database - have not looked in detail, but that's essentially what is happening here. I guess it can be fixed by changing the test code to not trigger DB operations in those tests (preferrable) or marking them as db_tests.

It makes sense, I'll check it out

**Review by potiuk — COMMENTED** [review]
I think we also need a newsfragment for that one. While it can be treated as 'security fix" it also potentially breaks somoene workflow if they have some non-CSRF logout integrated somewhere in their setup.

**Comment by shahar1** [comment]
> I think we also need a newsfragment for that one. While it can be treated as 'security fix" it also potentially breaks somoene workflow if they have some non-CSRF logout integrated somewhere in their setup.

I will add one.
I also checked locally - I managed to reproduce the errors, with or without the change, when running the following command:
`breeze testing non-db-tests --parallel-test-types "Always Providers[fab] WWW"`

So I see no good other choice, but marking `test_role_and_permission_endpoint.py` as a db test.

**Comment by potiuk** [comment]
> So I see no other good choice, but marking test_role_and_permission_endpoint.py as a db test.

Yep. Saw the marker. Indeed that's simpler.

File: airflow/www/templates/appbuilder/navbar_right.html (hunk 1 above)
Comment (bbovenzi) [reply airflow/www/templates/appbuilder/navbar_right.html:89]: A few css suggestions:

- Make the submit button a link:
 ```
<a
  href="javascript:{}"
  onclick="document.getElementById('logout').submit();" // add the id to the form
 >
  <span class="material-icons">exit_to_app</span>
  {{_("Log Out")}}
</a>
```
- You'll need to match the styles from `.dropdown-menu > li > a > .material-icons` in material-icons.css and from `.dropdown-menu > li > a` in `bootstrap-theme.css`

Comment (shahar1) [reply airflow/www/templates/appbuilder/navbar_right.html:89]: Thanks for the suggestion :) 
I didn't see your message in time, yet I implemented it using a `<button>` instead and matching the styles.
It seems to achieve the same effect:

image

**Review by potiuk — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Pull request merged**